sdk/scheduler/src/main/java/org/apache/mesos/offer/DefaultOfferRequirementProvider.java

```diff
@@ -102,27 +121,66 @@ public OfferRequirement getExistingOfferRequirement(PodInstance podInstance, Lis
+
+        LOGGER.info("Getting new TaskInfos for tasks: {}", tasksToLaunch);
+
+        ArrayList<String> usedResourceSets = new ArrayList<>();
         List<Protos.TaskInfo> taskInfos = new ArrayList<>();
+
+        LOGGER.info("Generating TaskInfos to launch.");
         for (TaskSpec taskSpec : podInstance.getPod().getTasks()) {
-            if (taskSpec.getGoal().equals(TaskSpec.GoalState.RUNNING)) {
-                taskInfos.add(getNewTaskInfo(podInstance, taskSpec));
+            if (!tasksToLaunch.contains(taskSpec.getName())) {
+                continue;
+            }
+
+            if (!usedResourceSets.contains(taskSpec.getResourceSet().getId())) {
+                LOGGER.info("Generating taskInfo for: {}", taskSpec.getName());
+                Protos.TaskInfo taskInfo = getNewTaskInfo(podInstance, taskSpec);
+                LOGGER.info("Using resource set: {}", taskSpec.getResourceSet().getId());
+                usedResourceSets.add(taskSpec.getResourceSet().getId());
+                taskInfos.add(taskInfo);
+            }
+        }
+
+        LOGGER.info("Generating TaskInfos to complete Pod footprint.");
+        for (TaskSpec taskSpec : podInstance.getPod().getTasks()) {
+            if (tasksToLaunch.contains(taskSpec.getName())) {
+                continue;
+            }
+
+            if (!usedResourceSets.contains(taskSpec.getResourceSet().getId())) {
+                LOGGER.info("Generating taskInfo for: {}", taskSpec.getName());
+                Protos.TaskInfo taskInfo = getNewTaskInfo(podInstance, taskSpec);
+                LOGGER.info("Marking as transient: {}", taskSpec.getName());
+                taskInfo = TaskUtils.setTransient(taskInfo);
+                LOGGER.info("Using resource set: {}", taskSpec.getResourceSet().getId());
+                usedResourceSets.add(taskSpec.getResourceSet().getId());
+                taskInfos.add(taskInfo);
             }
         }
 
         return taskInfos;
     }
 
-    private Protos.TaskInfo getNewTaskInfo(PodInstance podInstance, TaskSpec taskSpec)
-            throws InvalidRequirementException {
+    private Protos.TaskInfo getNewTaskInfo(
+            PodInstance podInstance,
+            TaskSpec taskSpec,
+            Collection<Protos.Resource> resources) throws InvalidRequirementException {
+
         Protos.TaskInfo.Builder taskInfoBuilder = Protos.TaskInfo.newBuilder()
                 .setName(TaskSpec.getInstanceName(podInstance, taskSpec))
                 .setTaskId(TaskUtils.emptyTaskId())
                 .setSlaveId(TaskUtils.emptyAgentId())
-                .addAllResources(getNewResources(taskSpec));
+                .addAllResources(resources);
 
-        TaskUtils.setTargetConfiguration(taskInfoBuilder, targetConfigurationId);
-        TaskUtils.setConfigFiles(taskInfoBuilder, taskSpec.getConfigFiles());
+        taskInfoBuilder = TaskUtils.setTargetConfiguration(taskInfoBuilder, targetConfigurationId);
```

Comment:
Totally agree.  Noticed this while working on it.  Filed INFINITY-762.  OfferRequirement property setting is a prereq of OfferEvaluator property setting right now.